From 8bc0dbeb6484b2d5bd74c23a081c575358cddaf2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20L=C3=BCpges?= <luepges@se-rwth.de>
Date: Fri, 22 May 2026 10:04:52 +0200
Subject: [PATCH] Future-port MCGrammarSymbolSurrogate replacements


diff --git a/monticore-generator/src/main/java/de/monticore/codegen/parser/MCGrammarInfo.java b/monticore-generator/src/main/java/de/monticore/codegen/parser/MCGrammarInfo.java
index 3e946d4b1..7d840fb21 100644
--- a/monticore-generator/src/main/java/de/monticore/codegen/parser/MCGrammarInfo.java
+++ b/monticore-generator/src/main/java/de/monticore/codegen/parser/MCGrammarInfo.java
@@ -10,7 +10,6 @@ import de.monticore.grammar.concepts.antlr.antlr._ast.ASTConceptAntlr;
 import de.monticore.grammar.concepts.antlr.antlr._ast.ASTJavaCodeExt;
 import de.monticore.grammar.grammar._ast.*;
 import de.monticore.grammar.grammar._symboltable.MCGrammarSymbol;
-import de.monticore.grammar.grammar._symboltable.MCGrammarSymbolSurrogate;
 import de.monticore.grammar.grammar._symboltable.ProdSymbol;
 import de.monticore.grammar.grammar._visitor.GrammarTraverser;
 import de.monticore.grammar.grammar._visitor.GrammarVisitor2;
@@ -265,8 +264,8 @@ public class MCGrammarInfo {
           ASTClassProd astProd = (ASTClassProd) ruleSymbol.getAstNode();
           if (astProd.getAltList().isEmpty()) {
             // if a rule has been overwritten and is empty, consider the superclass
-            for (MCGrammarSymbolSurrogate g : grammarSymbol.getSuperGrammars()) {
-              final Optional<ProdSymbol> ruleByName = g.lazyLoadDelegate().getProdWithInherited(astProd.getName());
+            for (MCGrammarSymbol g : grammarSymbol.getSuperGrammarSymbols()) {
+              final Optional<ProdSymbol> ruleByName = g.getProdWithInherited(astProd.getName());
               if (ruleByName.isPresent() && ruleByName.get().isClass()) {
                 if (ruleByName.get().isPresentAstNode() && ruleByName.get().getAstNode() instanceof ASTClassProd) {
                   astProd = (ASTClassProd) ruleByName.get().getAstNode();
diff --git a/monticore-generator/src/main/java/de/monticore/codegen/parser/ParserGeneratorHelper.java b/monticore-generator/src/main/java/de/monticore/codegen/parser/ParserGeneratorHelper.java
index 022693168..f7778dce6 100644
--- a/monticore-generator/src/main/java/de/monticore/codegen/parser/ParserGeneratorHelper.java
+++ b/monticore-generator/src/main/java/de/monticore/codegen/parser/ParserGeneratorHelper.java
@@ -14,7 +14,6 @@ import de.monticore.grammar.MCGrammarSymbolTableHelper;
 import de.monticore.grammar.PredicatePair;
 import de.monticore.grammar.grammar._ast.*;
 import de.monticore.grammar.grammar._symboltable.MCGrammarSymbol;
-import de.monticore.grammar.grammar._symboltable.MCGrammarSymbolSurrogate;
 import de.monticore.grammar.grammar._symboltable.ProdSymbol;
 import de.monticore.grammar.grammar_withconcepts.Grammar_WithConceptsMill;
 import de.monticore.grammar.grammar_withconcepts._ast.ASTAction;
@@ -555,8 +554,8 @@ public class ParserGeneratorHelper {
     if (!ast.getAltList().isEmpty()) {
       return ast.getAltList();
     }
-    for (MCGrammarSymbolSurrogate g : grammarSymbol.getSuperGrammars()) {
-      final Optional<ProdSymbol> ruleByName = g.lazyLoadDelegate().getProdWithInherited(ast.getName());
+    for (MCGrammarSymbol g : grammarSymbol.getSuperGrammarSymbols()) {
+      final Optional<ProdSymbol> ruleByName = g.getProdWithInherited(ast.getName());
       if (ruleByName.isPresent() && ruleByName.get().isClass()) {
         if (ruleByName.get().isPresentAstNode() && ruleByName.get().getAstNode() instanceof ASTClassProd) {
           return ((ASTClassProd)ruleByName.get().getAstNode()).getAltList();
--
2.45.1.windows.1

