From b6baf36a9e7b32cdf653b6444a2e1ebf87649de7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20L=C3=BCpges?= <luepges@se-rwth.de>
Date: Wed, 16 Jul 2025 16:46:16 +0200
Subject: [PATCH] initialize type system 3


diff --git a/monticore-generator/src/main/java/de/monticore/cli/MontiCoreTool.java b/monticore-generator/src/main/java/de/monticore/cli/MontiCoreTool.java
index b49061618..f9e97477a 100644
--- a/monticore-generator/src/main/java/de/monticore/cli/MontiCoreTool.java
+++ b/monticore-generator/src/main/java/de/monticore/cli/MontiCoreTool.java
@@ -13,7 +13,11 @@ import de.monticore.StatisticsHandlerFix;
 import de.monticore.cli.updateChecker.UpdateCheckerRunnable;
 import de.monticore.generating.templateengine.reporting.Reporting;
 import de.monticore.grammar.grammar_withconcepts.Grammar_WithConceptsMill;
-import de.monticore.io.FileReaderWriterFix;
+import de.monticore.grammar.grammar_withconcepts._visitor.Grammar_WithConceptsTraverser;
+import de.monticore.literals.mccommonliterals.types3.MCCommonLiteralsTypeVisitor;
+import de.monticore.types.mcbasictypes.types3.MCBasicTypesTypeVisitor;
+import de.monticore.types3.Type4Ast;
+import de.monticore.types3.util.MapBasedTypeCheck3;
 import de.se_rwth.commons.logging.Log;
 import de.se_rwth.commons.logging.Slf4jLog;
 import org.apache.commons.cli.*;
@@ -124,6 +128,8 @@ public class MontiCoreTool {
       
       // load custom or default script
       String script = loadScript(cmd);
+
+      initializeTypeCheck3();
       
       // execute MontiCore with loaded script and configuration
       new MontiCoreScript().run(script, configuration);
@@ -276,6 +282,26 @@ public class MontiCoreTool {
     return script;
   }
 
+  /**
+   * Initialize the TypeSystem(3)
+   */
+  public void initializeTypeCheck3() {
+    Grammar_WithConceptsTraverser traverser = Grammar_WithConceptsMill.traverser();
+
+    // map to store the results
+    Type4Ast type4Ast = new Type4Ast();
+
+    MCCommonLiteralsTypeVisitor visMCCommonLiterals = new MCCommonLiteralsTypeVisitor();
+    visMCCommonLiterals.setType4Ast(type4Ast);
+    traverser.add4MCCommonLiterals(visMCCommonLiterals);
+
+    MCBasicTypesTypeVisitor visMCBasicTypes = new MCBasicTypesTypeVisitor();
+    visMCBasicTypes.setType4Ast(type4Ast);
+    traverser.add4MCBasicTypes(visMCBasicTypes);
+
+    new MapBasedTypeCheck3(traverser, type4Ast).setThisAsDelegate();
+  }
+
   /*=================================================================*/
   /* Part 3: Defining the options incl. help-texts
   /*=================================================================*/
-- 
2.42.0.windows.2

