From 322ba8a376cda68535db6e9f89bc99fcc9c8c16a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20L=C3=BCpges?= <luepges@se-rwth.de>
Date: Thu, 17 Jul 2025 13:27:35 +0200
Subject: [PATCH] use inheritance traverser with cds


diff --git a/monticore-generator/src/main/java/de/monticore/codegen/cd2java/typecd2java/TypeCD2JavaDecorator.java b/monticore-generator/src/main/java/de/monticore/codegen/cd2java/typecd2java/TypeCD2JavaDecorator.java
index 464e8da50..aaf4164d4 100644
--- a/monticore-generator/src/main/java/de/monticore/codegen/cd2java/typecd2java/TypeCD2JavaDecorator.java
+++ b/monticore-generator/src/main/java/de/monticore/codegen/cd2java/typecd2java/TypeCD2JavaDecorator.java
@@ -17,7 +17,7 @@ public class TypeCD2JavaDecorator extends AbstractCreator<ASTCDCompilationUnit,
 
   @Override
   public ASTCDCompilationUnit decorate(final ASTCDCompilationUnit compilationUnit) {
-    CD4CodeTraverser traverser = CD4CodeMill.traverser();
+    CD4CodeTraverser traverser = CD4CodeMill.inheritanceTraverser();
     traverser.add4MCBasicTypes(new TypeCD2JavaVisitor(scope));
     compilationUnit.accept(traverser);
     return compilationUnit;
diff --git a/monticore-generator/src/main/java/de/monticore/codegen/mc2cd/MC2CD4CodeSymbolTableCompleter.java b/monticore-generator/src/main/java/de/monticore/codegen/mc2cd/MC2CD4CodeSymbolTableCompleter.java
index aba629201..704267049 100644
--- a/monticore-generator/src/main/java/de/monticore/codegen/mc2cd/MC2CD4CodeSymbolTableCompleter.java
+++ b/monticore-generator/src/main/java/de/monticore/codegen/mc2cd/MC2CD4CodeSymbolTableCompleter.java
@@ -14,7 +14,7 @@ public class MC2CD4CodeSymbolTableCompleter {
 
 
   public MC2CD4CodeSymbolTableCompleter() {
-    this.traverser = CD4CodeMill.traverser();
+    this.traverser = CD4CodeMill.inheritanceTraverser();
 
     //New!
     FullSynthesizeFromMCSGT4Grammar synthesize = new FullSynthesizeFromMCSGT4Grammar();
diff --git a/monticore-generator/src/main/java/de/monticore/codegen/mc2cd/manipul/JavaAndCdConformNameManipulation.java b/monticore-generator/src/main/java/de/monticore/codegen/mc2cd/manipul/JavaAndCdConformNameManipulation.java
index ef6ff639a..93b64ccf9 100644
--- a/monticore-generator/src/main/java/de/monticore/codegen/mc2cd/manipul/JavaAndCdConformNameManipulation.java
+++ b/monticore-generator/src/main/java/de/monticore/codegen/mc2cd/manipul/JavaAndCdConformNameManipulation.java
@@ -23,7 +23,7 @@ public class JavaAndCdConformNameManipulation implements UnaryOperator<ASTCDComp
 
   @Override
   public ASTCDCompilationUnit apply(ASTCDCompilationUnit cdCompilationUnit) {
-    CD4AnalysisTraverser traverser = CD4AnalysisMill.traverser();
+    CD4AnalysisTraverser traverser = CD4AnalysisMill.inheritanceTraverser();
     traverser.add4CDBasis(new ManipulateVisitor(cdCompilationUnit.getCDPackageList(), cdCompilationUnit.getCDDefinition().getName()));
     cdCompilationUnit.accept(traverser);
     return cdCompilationUnit;
-- 
2.45.1.windows.1

